src/main/java/de/retest/recheck/report/TestReportFilter.java

```diff
@@ -45,7 +46,11 @@ static TestReplayResult filter( final TestReplayResult testReplayResult, final F
 		final TestReplayResult newTestReplayResult =
 				new TestReplayResult( testReplayResult.getName(), testReplayResult.getTestNr() );
 		for ( final ActionReplayResult actionReplayResult : testReplayResult.getActionReplayResults() ) {
-			newTestReplayResult.addAction( filter( actionReplayResult, filter ) );
+			if ( actionReplayResult instanceof NoGoldenMasterActionReplayResult ) {
```

Comment:
Isn't the actual problem that `filter( ActionReplayResult, Filter )` filters `NoGoldenMasterActionReplayResult`, which it shouldn't?

Reply:
Good point